app/src/main/java/com/lastaoutdoor/lasta/viewmodel/MapViewModel.kt

```diff
@@ -58,51 +57,40 @@ constructor(private val outdoorActivityRepository: OutdoorActivityRepository) :
   private fun fetchClimbingActivities(
       rad: Double,
       centerLocation: LatLng,
-      repository: OutdoorActivityRepository
+      repository: ActivityRepository
   ) {
 
-    // list that will store the climbing activities nodes
-    var climbingNodes: List<Node> = emptyList()
-
-    // Since the API call is a network call, it needs to be done in a separate thread
-    val climbingThread = Thread {
-      climbingNodes =
-          repository
-              .getClimbingActivitiesNode(
-                  rad.toInt(), // in meters
-                  centerLocation.latitude,
-                  centerLocation.longitude)
-              .elements
-    }
+    viewModelScope.launch {
```

Comment:
Nice! I also thought that we could possibly change the Thread call to a viewModelScope Launch

Reply:
Thank you, I mean most of your UI had already implemented text that I could just insert the data in. It was pretty easy to integrate thanks to your upfront work!